### ISSUE TITLE
Creating a kinesis firehose delivery stream fails with an rpc error

### ISSUE THREAD

**Issue by nickpwhite** [terraform-providers/terraform-provider-aws#12600]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```bash
Terraform v0.12.24
+ provider.aws v2.55.0
```

### Affected Resource(s)

* aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files

```hcl
provider "aws" {
  allowed_account_ids = ["#########"]
  region              = "us-east-1"
  version             = "2.55.0"
}

terraform {
  required_version = "=0.12.24"
}

data "aws_s3_bucket" "datalytics" {
  bucket = "datalytics.data"
}

resource "aws_iam_role" "firehose" {
  name               = "qa_firehose_delivery"
  assume_role_policy = <<-EOF
  {
    "Version": "2012-10-17",
    "Statement": [
      {
        "Action": "sts:AssumeRole",
        "Principal": {
          "Service": "firehose.amazonaws.com"
        },
        "Effect": "Allow",
        "Sid": ""
      }
    ]
  }
  EOF
}

resource "aws_glue_catalog_database" "ses_records_database" {
  name = "qa__ses_records"
}

resource "aws_glue_catalog_table" "ses_records_table" {
  name = "ses_records"
  database_name = aws_glue_catalog_database.ses_records_database.name
}

resource "aws_kinesis_firehose_delivery_stream" "event_stream" {
  name        = "qa_ses_records_firehose"
  destination = "extended_s3"

  extended_s3_configuration {
    role_arn            = aws_iam_role.firehose.arn
    bucket_arn          = data.aws_s3_bucket.datalytics.arn
    buffer_interval     = "300"
    buffer_size         = "128"
    compression_format  = "GZIP"

    data_format_conversion_configuration {
      input_format_configuration {
        deserializer {
          open_x_json_ser_de {}
        }
      }

      output_format_configuration {
        serializer {
          parquet_ser_de {}
        }
      }

      schema_configuration {
        database_name = aws_glue_catalog_database.ses_records_database.name
        role_arn      = aws_iam_role.firehose.arn
        table_name    = aws_glue_catalog_table.ses_records_table.name
      }
    }

    processing_configuration {}

    s3_backup_mode = "Enabled"

    s3_backup_configuration {
      bucket_arn = data.aws_s3_bucket.datalytics.arn
      role_arn   = aws_iam_role.firehose.arn
    }
  }
}
```

### Debug Output
[debug.log](https://gist.github.com/nickpwhite/2e2838edced38c852a0230763c22acaa)

### Panic Output
[crash.log](https://gist.github.com/nickpwhite/e9e1e18d9bee9c2a398ee20e248e21e0)

### Expected Behavior
The `aws_kinesis_firehose_delivery_stream` was created along with the 3 other resources.

### Actual Behavior
Only the `aws_iam_role`, `aws_glue_catalog_database`, and `aws_glue_catalog_table` were created. Terraform panicked trying to create the `aws_kinesis_firehose_delivery_stream`.

### Steps to Reproduce
1. Create an S3 bucket called "datalytics.data"
2. `terraform init`
3. `terraform apply`

**Comment by bflad** [comment]
Hi @nickpwhite 👋 Thank you for submitting this and sorry you ran into trouble here.

To fix this in your configuration, you should be able to not generate a panic by removing the following:

```hcl
resource "aws_kinesis_firehose_delivery_stream" "event_stream" {
  # ... other configuration ...
  extended_s3_configuration {
    # ... remove below ...
    processing_configuration {}
    # ... remove above ...
  }
}
```

The code stacktrace for this is the following:

```
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: panic: interface conversion: interface {} is nil, not map[string]interface {}
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: 
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: goroutine 88 [running]:
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.extractProcessingConfiguration(0xc001097860, 0xc001097860)
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:1740 +0x2ef
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.createExtendedS3Config(0xc0003d25b0, 0x58fa1da)
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:1447 +0x877
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsKinesisFirehoseDeliveryStreamCreate(0xc0003d25b0, 0x4e24fc0, 0xc000300500, 0x2, 0x96ee420)
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream.go:2119 +0xeec
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000578380, 0xc001003040, 0xc00089f160, 0x4e24fc0, 0xc000300500, 0xc000661a01, 0xc00113f7d0, 0xc000661b20)
2020-03-31T13:47:33.683-0700 [DEBUG] plugin.terraform-provider-aws_v2.55.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:305 +0x365
```

Which aligns with this code: https://github.com/terraform-providers/terraform-provider-aws/blob/c46c4dcb154a5e81238e2c0bcccc9391c4429ee3/aws/resource_aws_kinesis_firehose_delivery_stream.go#L1740

Introducing a `nil` check a few lines up where `len(config) == 0` is checked, e.g. `len(config) == 0 || config[0] == nil` would prevent this in the code.

**Comment by bflad** [comment]
Fix submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/12613

**Comment by bflad** [comment]
The fix to prevent the panic in this situation has been merged and will release with version 2.60.0, later this week.
